packages/core/src/tools/utils.ts

```diff
@@ -146,61 +146,55 @@ export function round(num: number, decimals: 0 | 1 | 2 | 3 | 4) {
 // eslint-disable-next-line @typescript-eslint/no-empty-function
 export function noop() {}
 
-interface ObjectWithToJSON {
-  toJSON: (() => object) | undefined
-}
-
-type OriginalToJSON = [boolean, undefined | (() => object)]
-
 /**
- * Custom implementation of JSON.stringify that ignores value.toJSON.
- * We need to do that because some sites badly override toJSON on certain objects.
- * Note this still supposes that JSON.stringify is correct...
+ * Custom implementation of JSON.stringify that ignores some toJSON methods. We need to do that
+ * because some sites badly override toJSON on certain objects. Removing all toJSON methods from
+ * nested values would be too costly, so we just detach them from the root value, and native classes
+ * used to build JSON values (Array and Object).
+ *
+ * Note: this still assumes that JSON.stringify is correct.
  */
 export function jsonStringify(
   value: unknown,
   replacer?: Array<string | number>,
   space?: string | number
 ): string | undefined {
-  if (value === null || value === undefined) {
+  if (typeof value !== 'object' || value === null) {
     return JSON.stringify(value)
   }
-  let originalToJSON: OriginalToJSON = [false, undefined]
-  if (hasToJSON(value)) {
-    // We need to add a flag and not rely on the truthiness of value.toJSON
-    // because it can be set but undefined and that's actually significant.
-    originalToJSON = [true, value.toJSON]
-    delete value.toJSON
-  }
 
-  let originalProtoToJSON: OriginalToJSON = [false, undefined]
-  let prototype
-  if (typeof value === 'object') {
-    prototype = Object.getPrototypeOf(value) as object
-    if (hasToJSON(prototype)) {
-      originalProtoToJSON = [true, prototype.toJSON]
-      delete prototype.toJSON
-    }
-  }
+  // Note: The order matter here. We need to detach toJSON methods on parent classes before their
+  // subclasses.
+  const restoreObjectPrototypeToJson = detachToJsonMethod(Object.prototype)
+  const restoreArrayPrototypeToJson = detachToJsonMethod(Array.prototype)
+  const restoreValuePrototypeToJson = detachToJsonMethod(Object.getPrototypeOf(value))
+  const restoreValueToJson = detachToJsonMethod(value)
```

Comment:
@BenoitZugmeyer could you please clarify while all objects are stripped of toJSON method before adding it to logs? Many classes because of it are rendered simply as `{}` which isn't very convenient and now needs extra logic to invoke `toJSON` on those objects _before_ passing it to the DataDog library.

Any chance it will be made configurable?

Thank you

Reply:
Hi @ArmorDarks, can you please open an issue to start a new discussion?

Reply:
We found out that some custom `toJSON` implementations were throwing exceptions and broke everything. That's why we are removing those. But we are in the process of re-designing how we sanitize/serialize user data, and we are talking of respecting the `toJSON` methods (behind a try/catch)